Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarQubeWebServerTest.cs

```diff
@@ -28,532 +28,550 @@
-        }
+        Func<Task> action = async () => await service.DownloadProperties(ProjectKey, null);
 
-        [TestMethod]
-        public void DownloadProperties_Sq63plus_Forbidden()
-        {
-            var downloaderMock = Mock.Of<IDownloader>(x => x.TryDownloadIfExists(It.IsAny<string>(), It.IsAny<bool>()) == Task.FromException<Tuple<bool, string>>(new HttpRequestException("Forbidden")));
-            var service = new SonarQubeWebServer(downloaderMock, new Version("6.3.0.0"), logger, null);
+        await action.Should().ThrowAsync<HttpRequestException>();
+    }
 
-            Action action = () => _ = service.DownloadProperties(ProjectKey, null).Result;
+    [TestMethod]
+    public void DownloadProperties_Sq63plus_Forbidden()
+    {
+        var downloaderMock = Substitute.For<IDownloader>();
+        downloaderMock.TryDownloadIfExists(Arg.Any<string>(), Arg.Any<bool>())
+            .Returns(Task.FromException<Tuple<bool, string>>(new HttpRequestException("Forbidden")));
+        var service = new SonarQubeWebServer(downloaderMock, new Version("6.3.0.0"), logger, null);
 
-            action.Should().Throw<HttpRequestException>();
-        }
+        Action action = () => _ = service.DownloadProperties(ProjectKey, null).Result;
 
-        [TestMethod]
-        public async Task DownloadProperties_SQ63AndHigherWithProject_ShouldBeEmpty()
-        {
-            var downloaderMock = new Mock<IDownloader>();
-            downloaderMock.Setup(x => x.TryDownloadIfExists(It.IsAny<string>(), It.IsAny<bool>())).ReturnsAsync(Tuple.Create(true, "{ settings: [ ] }")).Verifiable();
-            sut = new SonarQubeWebServer(downloaderMock.Object, new Version("6.3"), logger, null);
+        action.Should().Throw<HttpRequestException>();
+    }
 
-            var properties = await sut.DownloadProperties("key", null);
+    [TestMethod]
+    public async Task DownloadProperties_SQ63AndHigherWithProject_ShouldBeEmpty()
+    {
+        var downloaderMock = Substitute.For<IDownloader>();
+        downloaderMock.TryDownloadIfExists(Arg.Any<string>(), Arg.Any<bool>()).Returns(Task.FromResult(Tuple.Create(true, "{ settings: [ ] }")));
+        sut = new SonarQubeWebServer(downloaderMock, new Version("6.3"), logger, null);
 
-            properties.Should().BeEmpty();
-            downloaderMock.Verify();
-        }
+        var properties = await sut.DownloadProperties("key", null);
 
-        [TestMethod]
-        public async Task DownloadProperties_OlderThanSQ63_ShouldBeEmpty()
-        {
-            var downloaderMock = new Mock<IDownloader>();
-            downloaderMock.Setup(x => x.Download(It.IsAny<string>(), It.IsAny<bool>())).ReturnsAsync("[]").Verifiable();
-            sut = new SonarQubeWebServer(downloaderMock.Object, new Version("6.2.9"), logger, null);
+        properties.Should().BeEmpty();
+        await downloaderMock.Received().TryDownloadIfExists(Arg.Any<string>(), Arg.Any<bool>());
```

Comment:
Why not assert the concrete values, to demonstrate what really happened? Like above with the `Download` call?